compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/descriptors/IrDelegateDescriptor.kt

```diff
@@ -95,6 +96,9 @@ class IrPropertyDelegateDescriptorImpl(
 ) :
     IrDelegateDescriptorBase(
         correspondingProperty.containingDeclaration,
+        // Inherit the source node so that the parent class of package-level
```

Comment:
I would prefer to patch KotliTypeMapper.internalNameForPackageMemberOwner
```
if (isIrBackend && directMember is IrPropertyDelegateDescriptor) {
                return internalNameForPackageMemberOwner(directMember.correspondingProperty, publicFacade, isIrBackend)
            }
```

With some time there would be IR-specific KotliTypeMapper without any hacks